Sources/SwiftInspectorVisitors/PropertyVisitor.swift

```diff
@@ -58,6 +58,36 @@ public final class PropertyVisitor: SyntaxVisitor {
     return .skipChildren
   }
 
+  public override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    if visitTypeDeclarationChildren {
+      return .visitChildren
+    } else {
+      // We've encountered a nested type declaration.
+      // Skip it to avoid scanning a child type's properties.
+      return .skipChildren
+    }
+  }
+
+  public override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+    if visitTypeDeclarationChildren {
+      return .visitChildren
+    } else {
+      // We've encountered a nested type declaration.
+      // Skip it to avoid scanning a child type's properties.
+      return .skipChildren
+    }
+  }
+
+  public override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+    if visitTypeDeclarationChildren {
+      return .visitChildren
+    } else {
+      // We've encountered a nested type declaration.
+      // Skip it to avoid scanning a child type's properties.
+      return .skipChildren
+    }
+  }
```

Comment:
Thoughts on whether I should also stop on `Protocol` and `Extension` syntax nodes? Those aren't nestable, but they are type declarations... aaaaand I just talked myself into it more commits coming.